source/Octopus.Tentacle/Kubernetes/KubernetesPodMonitorTask.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Diagnostics;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public interface IKubernetesPodMonitorTask
+    {
+        void Start();
+        void Stop();
+    }
+
+    public class KubernetesPodMonitorTask: IKubernetesPodMonitorTask, IDisposable
+    {
+        readonly IKubernetesPodMonitor podMonitor;
+        readonly ISystemLog log;
+        readonly CancellationTokenSource cancellationTokenSource = new ();
+
+        static readonly object LockObj = new();
```

Comment:
nit: this probably shouldn't be static, it's there to protect instance properties so it should also be an instance property. If you had two instances of `KubernetesPodMonitorTask` they would share the same lock. I don't think this is really an issue though as it's registered as a singleton, thus a "nit" 😅.

Reply:
fair call, I'll change it :)